python/ql/lib/semmle/python/dataflow/new/internal/TypeTrackingImpl.qll

```diff
@@ -181,46 +182,50 @@ module TypeTrackingInput implements Shared::TypeTrackingInput {
    * Holds if `nodeFrom` is being written to the `content` content of the object in `nodeTo`.
    */
   predicate storeStep(Node nodeFrom, Node nodeTo, Content content) {
-    exists(DataFlowPublic::AttrWrite a |
-      a.mayHaveAttributeName(content) and
+    exists(DataFlowPublic::AttrWrite a, string attrName |
+      content.(DataFlowPublic::AttributeContent).getAttribute() = attrName and
+      a.mayHaveAttributeName(attrName) and
       nodeFrom = a.getValue() and
       nodeTo = a.getObject()
     )
```

Comment:
This looks like it could be subsumed by `attributeStoreStep` (which could be part of `storeStepCommon`), if we are willing to replace `mayHaveAttributeName` with the slightly stricter `getAttributeName`. Do we remember why we are strict using dataflow and not using type tracking? Is it simply that we do not yet (as in evaluation order) have access to this small local flow reasoning during dataflow?

Reply:
I think it would be interesting to be able to share this part as well between dataflow and type-tracking, but for this PR I didn't want to deal with that complexity, so I just kept the existing behavior.